db/compaction/compaction_iterator.cc

```diff
@@ -713,7 +704,7 @@ void CompactionIterator::NextFromInput() {
             // outputted on the next iteration.)
 
             // Setting valid_ to true will output the current SingleDelete
-            valid_ = true;
+            validity_info_.SetValid(ValidContext::kKeepSDForCC);
```

Comment:
Some of these names are not entirely obvious; how about `kKeepSDForConflictChecking`?

Reply:
Will improve the naming.